### ISSUE TITLE
Networking Updates

### ISSUE THREAD

**Issue by JonathanLindsey** [wpilibsuite/frc-docs#493]
I am putting together updates for the networking section. For example, as of today (1/20/2020) the [Mixing Dynamic and Static Configurations](http://docs.wpilib.org/en/latest/docs/networking/networking-introduction/networking-basics.html#mixing-dynamic-and-static-configurations) section is outdated.
>In the pits, a team may encounter issues with mixing Static and DHCP devices for the following reason. As mentioned above, DHCP devices will fall back to a link-local address (169.254.xx.yy) if a server isn’t present.

The robot radio now acts as a DHCP server, therefore devices will not fall back to link local in the pits unless the robot radio is disconnected.

Anyway, does anyone have more editable versions of the the drawings like this one?
![image](https://user-images.githubusercontent.com/18510187/72778135-97464f00-3bdd-11ea-8910-a26ee08ba45b.png)
Just a source for the device pictures would be great.

**Comment by Daltz333** [comment]
Unfortunately the source for the images is no longer available.

**Comment by yellowjaguar5** [comment]
I'm working on re-doing all of the frc-docs diagrams in PR #620. I replicated the diagram currently shown in the docs below. Please let me know what is inaccurate and I would be happy to adjust accordingly. @JonathanLindsey @Daltz333 

---

![image](https://user-images.githubusercontent.com/15172054/79509837-29010500-7ff1-11ea-93ba-43ece98deba8.png)

**Comment by AustinShalit** [comment]
Funny enough... that graphic is out of date. The robot radio now runs a DHCP server on the ethernet side of the network meaning all devices should never get a local link IP address.
From https://docs.wpilib.org/en/stable/docs/networking/networking-introduction/ip-configurations.html:

- OpenMesh OM5P-AN or OM5P-AC radio - Static 10.TE.AM.1 programmed by Kiosk
- roboRIO - DHCP 10.TE.AM.2 assigned by the Robot Radio
- Driver Station - DHCP 10.TE.AM.X assigned by field when on the field; assigned by robot radio in the pits
- IP camera (if used) - DHCP 10.TE.AM.Y assigned by Robot Radio
- Other devices (if used) - DHCP 10.TE.AM.Z assigned by Robot Radio

**Comment by yellowjaguar5** [comment]
Sounds good. @AustinShalit, I thought that IP cameras and other devices will not be accessible in the pits. Is this still true?

**Comment by prateekma** [comment]
These devices should be accessible in the pits from the DS Laptop if you connect via the robot radio.

**Comment by yellowjaguar5** [comment]
Thanks for the correction. How does this revision look? 
![Screen Shot 2020-04-16 at 4 58 24 PM](https://user-images.githubusercontent.com/15172054/79517597-8736e380-8003-11ea-8780-e2930eece0d6.png)

**Comment by AustinShalit** [comment]
That looks good. What if you changed the field diagram to have a dashed line between the DS and bridge to highlight that it is not a direct connection (it really goes through a few switches and a AP).

**Comment by AustinShalit** [comment]
You might add a third panel that shows connection via USB.

**Comment by yellowjaguar5** [comment]
Good idea. I'm not sure how well this image fits in its original place [here](https://docs.wpilib.org/en/stable/docs/networking/networking-introduction/networking-basics.html#mixing-dynamic-and-static-configurations). Thoughts on placement?Anything else on the diagram?
![image](https://user-images.githubusercontent.com/15172054/79524842-5c09bf80-8016-11ea-9c95-674da2051070.png)

**Comment by AustinShalit** [comment]
1. On the field - good
2. In the pits via radio - good
3. In the pits via USB - roboRIO gets 2 IP addresses here: the 10.TE.AM.2 on the "green" network and 172.22.11.2 on the "blue" network. Driver station does not get a 10.TE.AM.X address, it gets a 172.22.11.1 (I think, need to check). We should highlight that without a network forwarder on the rio, devices on the "green" network cannot talk to devices on the "blue" network

Might consider swapping the IP cameras for a raspberry PIs because that is what most teams use these days.

**Comment by sciencewhiz** [comment]
a network forwarder like https://docs.wpilib.org/en/stable/docs/networking/networking-utilities/portforwarding.html

**Comment by yellowjaguar5** [comment]
Sounds good. Latest revision below. I'll add a note in the article that a network forwarder is required to bridge the blue and green networks when connected via USB.
![image](https://user-images.githubusercontent.com/15172054/79530107-3b943200-8023-11ea-8b8c-e6970f0dbac7.png)

**Comment by Daltz333** [comment]
Just a small nit pick, could the inside corners not be rounded?

**Comment by yellowjaguar5** [comment]
As requested:
![image](https://user-images.githubusercontent.com/15172054/79530754-1ef8f980-8025-11ea-9bc8-3ed1dc3ea3aa.png)

**Comment by JonathanLindsey** [comment]
Thanks for redoing that diagram and all the others @yellowjaguar5.

I noticed that the Driver Station icon is broken when rendered in FireFox and Chrome. However, it is not broken when opened in the online version of diagrams.net. Do you have any idea why that would be and how to fix it?
https://docs.wpilib.org/en/latest/docs/networking/networking-introduction/networking-basics.html#mixing-dynamic-and-static-configurations

**Comment by yellowjaguar5** [comment]
Thanks for the ping. School is just starting for me, so I have very little time for side projects. I hope to address it in the next two weeks. Future folks who stumble across this, feel free to ping me again if I don't get to it in a timely manner.

**Comment by JonathanLindsey** [comment]
No problem.
Actually I re-exported it from app.diagrams.net and it appears to be working now. Still not sure why it happened though.

**Comment by yellowjaguar5** [comment]
@JonathanLindsey I created a new PR (#820) to address this issue. Can you verify that [this diagram](https://frc-docs--820.org.readthedocs.build/en/820/docs/networking/networking-introduction/networking-basics.html#mixing-dynamic-and-static-configurations) has all the icons present now? 

@Daltz333 Can this issue be closed once #820 is merged?

**Comment by JonathanLindsey** [comment]
@yellowjaguar5 I just checked it by using the above link now that it's merged and it works. Thanks.
